### ISSUE TITLE
Rule request: forbid return in finally

### ISSUE THREAD

**Issue by louy** [palantir/tslint#1097]
Adding a `return` statement in a `finally` block is a very confusing thing. Thrown errors get ignored, `return` statements in `try` blocks get ignored, and your mind explodes. 

I'd love to have a rule to forbid this. Especially now that we're having `async`/`await`.

examples:

``` js
try {
return 1;
} catch(err) {
return 2;
} finally {
return 3;
}
// > 3

try {
throw new Error;
return 1;
} catch(err) {
return 2;
} finally {
return 3;
}
// > 3
```

**Comment by adidahiya** [comment]
sounds reasonable. is there any analogous rule for ESLint? if not, that's cool too

**Comment by louy** [comment]
not yet. https://github.com/eslint/eslint/issues/5808

**Comment by andy-hanson** [comment]
Should be fixed by #1349.
